### PR TITLE
initialize welcome webview

### PR DESCRIPTION
**Pull request by Eskibear**
See latest comments for demo

### CODE REVIEW

File: src/welcome/assets/components/QuickActionPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import * as React from 'react';
+import { ListGroup } from 'react-bootstrap';
+import { encodeCommandUri } from '../utils';
+
+export default class QuickActionPanel extends React.Component {
+    render() {
+        const actions = [
+            { name: "Create a New Project", command: "java.project.create" },
+            { name: "Open an Existing Project", command: "workbench.action.files.openFolder", os: "win" },
+            // { name: "Open an Existing Project", command: "workbench.action.files.openFileFolder", os: "mac" },
```
Comment (jdneo) [inline src/welcome/assets/components/QuickActionPanel.tsx:13]: Is it possible to check if the command `workbench.action.files.openFileFolder` is available or not and then decide which command to call?

Comment (Eskibear) [reply src/welcome/assets/components/QuickActionPanel.tsx:13]: I'm also thinking of it. Looks like `data-os` is used in `<a>` nodes in "Overview" page. I also tried but it didn't work.

**Comment by Eskibear** [comment]
image

**Comment by Eskibear** [comment]
Finalized Welcome view:
image

File: src/welcome/assets/components/TourPage.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            title: "Testing",
+            description: "Use the Testing view to run unit tests.",
+            imageUri: require("../resources/testing.png")
+        }];
+
+        return content.map((elem) => <div className="text-center">
+            <h2>{elem.title}</h2>
+            <div>{elem.description}</div>
+            <img src={elem.imageUri} alt={elem.title} />
+            <Button onClick={this.nextStep}>Next Step</Button>
```
Comment (jdneo) [inline src/welcome/assets/components/TourPage.tsx:97]: How about adding `Previous Step`?

File: src/welcome/assets/components/TourPage.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            <div><a href="#" onClick={() => showWelcomePage(false)}>skip</a></div>
+        </div>;
+    }
+
+    getEndingPage = () => {
+        return <div>
+            <img src={doneIcon} alt="logo" className="logo"></img>
+            <h2>You’re good to go!</h2>
+            <div>Next, start using Java!</div>
+            <div><Button onClick={this.nextStep}>What next?</Button></div>
```
Comment (jdneo) [inline src/welcome/assets/components/TourPage.tsx:117]: What's next?

File: src/welcome/index.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import * as path from "path";
+import { loadTextFromFile } from "../utils";
+import { instrumentSimpleOperation } from "vscode-extension-telemetry-wrapper";
+
+const KEY_SHOW_WHEN_USING_JAVA = "showWhenUsingJava";
+let welcomeView: vscode.WebviewPanel | undefined;
+
+export async function showWelcomeWebview(context: vscode.ExtensionContext, args?: any[]) {
+    if (welcomeView) {
+        const firstTimeRun = args![0]!.firstTimeRun;
```
Comment (jdneo) [inline src/welcome/index.ts:14]: Use `?.` to avoid NPE?

File: src/welcome/index.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                        }
+                    });
+                }
+            default:
+                break;
+        }
+    })));
+
+    let firstTimeRun;
+    if (args && args[0] && args[0].firstTimeRun !== undefined) {
```
Comment (jdneo) [inline src/welcome/index.ts:65]: `if (args?.[0]?.firstTimeRun)` ?

File: src/welcome/assets/components/TourPage.tsx (hunk 3 above)
Comment (Eskibear) [reply src/welcome/assets/components/TourPage.tsx:117]: got it

File: src/welcome/assets/components/TourPage.tsx (hunk 2 above)
Comment (Eskibear) [reply src/welcome/assets/components/TourPage.tsx:97]: I'll make the page indicator (dots below) clickable.

**Review by jdneo — APPROVED** [review]

**Pull request merged**